### ISSUE TITLE
Segfault in AvroRowInputFormat

### ISSUE THREAD

**Issue by tavplubix** [ClickHouse/ClickHouse#29218]
```
01543_avro_deserialization_with_lc:                                     [ FAIL ] - having stderror: 
/home/avtokmakov/ch/ClickHouse/tests/queries/0_stateless/01543_avro_deserialization_with_lc.sh: line 18: 190694 Done                    $CLICKHOUSE_CLIENT -q "SELECT * FROM test_01543 FORMAT Avro"
     190695 Aborted                 (core dumped) | $CLICKHOUSE_CLIENT -q "INSERT INTO test_01543 FORMAT Avro"
[dlat5511] 2021.09.21 13:26:47.026939 [ 190712 ] <Fatal> BaseDaemon: ########################################
[dlat5511] 2021.09.21 13:26:47.027331 [ 190712 ] <Fatal> BaseDaemon: (version 21.11.1.1, build id: DC8DEE7609503E55) (from thread 190477) (query_id: 465129a8-f27a-40c8-8606-8b44b949a8d1) Received signal Segmentation fault (11)
[dlat5511] 2021.09.21 13:26:47.027776 [ 190712 ] <Fatal> BaseDaemon: Address: 0x7f4f7bfbbe80 Access: write. Attempted access has violated the permissions assigned to the memory area.
[dlat5511] 2021.09.21 13:26:47.028157 [ 190712 ] <Fatal> BaseDaemon: Stack trace: 0x204750a6 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4 0x20475fa4
[dlat5511] 2021.09.21 13:26:47.339041 [ 190712 ] <Fatal> BaseDaemon: 4. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:180: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x204750a6 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:47.577315 [ 190712 ] <Fatal> BaseDaemon: 5. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:47.815728 [ 190712 ] <Fatal> BaseDaemon: 6. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:48.078003 [ 190712 ] <Fatal> BaseDaemon: 7. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:48.364665 [ 190712 ] <Fatal> BaseDaemon: 8. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:48.653125 [ 190712 ] <Fatal> BaseDaemon: 9. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:48.938051 [ 190712 ] <Fatal> BaseDaemon: 10. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:49.226233 [ 190712 ] <Fatal> BaseDaemon: 11. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:49.512380 [ 190712 ] <Fatal> BaseDaemon: 12. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:49.797789 [ 190712 ] <Fatal> BaseDaemon: 13. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:50.084191 [ 190712 ] <Fatal> BaseDaemon: 14. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:50.371789 [ 190712 ] <Fatal> BaseDaemon: 15. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:50.655531 [ 190712 ] <Fatal> BaseDaemon: 16. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:50.934311 [ 190712 ] <Fatal> BaseDaemon: 17. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:51.226836 [ 190712 ] <Fatal> BaseDaemon: 18. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:51.511977 [ 190712 ] <Fatal> BaseDaemon: 19. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:51.802979 [ 190712 ] <Fatal> BaseDaemon: 20. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:52.101297 [ 190712 ] <Fatal> BaseDaemon: 21. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:52.391579 [ 190712 ] <Fatal> BaseDaemon: 22. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:52.679386 [ 190712 ] <Fatal> BaseDaemon: 23. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:52.954888 [ 190712 ] <Fatal> BaseDaemon: 24. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:53.221894 [ 190712 ] <Fatal> BaseDaemon: 25. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:53.498462 [ 190712 ] <Fatal> BaseDaemon: 26. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:53.777213 [ 190712 ] <Fatal> BaseDaemon: 27. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:54.062786 [ 190712 ] <Fatal> BaseDaemon: 28. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:54.346952 [ 190712 ] <Fatal> BaseDaemon: 29. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:54.633110 [ 190712 ] <Fatal> BaseDaemon: 30. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:54.916362 [ 190712 ] <Fatal> BaseDaemon: 31. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:55.204293 [ 190712 ] <Fatal> BaseDaemon: 32. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:55.486708 [ 190712 ] <Fatal> BaseDaemon: 33. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:55.772645 [ 190712 ] <Fatal> BaseDaemon: 34. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:56.064206 [ 190712 ] <Fatal> BaseDaemon: 35. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:56.352559 [ 190712 ] <Fatal> BaseDaemon: 36. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:56.632029 [ 190712 ] <Fatal> BaseDaemon: 37. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:56.871084 [ 190712 ] <Fatal> BaseDaemon: 38. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:57.111300 [ 190712 ] <Fatal> BaseDaemon: 39. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:57.350727 [ 190712 ] <Fatal> BaseDaemon: 40. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:57.590458 [ 190712 ] <Fatal> BaseDaemon: 41. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:57.828643 [ 190712 ] <Fatal> BaseDaemon: 42. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:58.070135 [ 190712 ] <Fatal> BaseDaemon: 43. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:58.310467 [ 190712 ] <Fatal> BaseDaemon: 44. /home/avtokmakov/ch/ClickHouse/build_debug/../src/Processors/Formats/Impl/AvroRowInputFormat.cpp:382: DB::AvroDeserializer::createDeserializeFn(std::__1::shared_ptr<avro::Node>, std::__1::shared_ptr<DB::IDataType const>) @ 0x20475fa4 in /home/avtokmakov/ch/bin/tmp/clickhouse
[dlat5511] 2021.09.21 13:26:59.069424 [ 190712 ] <Fatal> BaseDaemon: Calculated checksum of the binary: 808C424016C3E9426E0F6FE092DDA021. There is no information about the reference checksum.
Code: 32. DB::Exception: Attempt to read after eof: while receiving packet from localhost:9000. (ATTEMPT_TO_READ_AFTER_EOF)

Code: 210. DB::NetException: Connection refused (localhost:9000). (NETWORK_ERROR)



stdout:
0	0
1	1
2	2
3	3
4	4
5	5
6	6
7	7
8	8
9	9

```

**Comment by alexey-milovidov** [comment]
@Avogar This is major issue.

**Comment by Avogar** [comment]
@tavplubix can you attach the fuzzer report?

**Comment by tavplubix** [comment]
It's not from fuzzer, I got it when running functional tests locally. But possibly tests were newer than binary (and the bug is already fixed by #29132)

**Comment by Avogar** [comment]
> But possibly tests were newer than binary 
I think this caused the problem. The test was updated in https://github.com/ClickHouse/ClickHouse/pull/29132 to check found segfault and it works on master. I ran this test on binary without my fix and get exactly the same segfault and stacktrace. So, I think there is no problem here.
